go-controller/pkg/ovn/egressip.go

```diff
@@ -2203,6 +2218,9 @@ func checkEgressNodesReachabilityIterate(oc *Controller) {
 				reAddOrDelete[eNode.name] = false
 			}
 			eNode.isReachable = isReachable
+		} else {
+			// End connection (if there is one).
+			eNode.healthClient.Disconnect()
```

Comment:
is this Disconnect important? I was trying to figure out why it was necessary. Is it just because we know the node is not ready or has no label, we can assume its not ready yet to be checked so disconnect?

Reply:
That is right @trozet . So what happens now is that EgressIP will call isReady on _all_ nodes _once_ upon starting. Then, it will probe monitor only the nodes labelled `k8s.ovn.org/egress-assignable`. Without this change, we would end up with an open gRPC connection to all nodes, including ones that did not have the label, or had its label removed later on. Since calling Disconnect is cheap for nodes that not connected, I'm using this this codepath as the one place where non-monitored nodes have no open gRPC connection. I will add a comment in the code for that.

Reply:
This is the call stack that checks if node is reachable, regardless of the label:
```
func (oc *Controller) addResource(objectsToRetry *RetryObjs, obj interface{}, fromRetryLoop bool) error {
==>
 case factory.EgressNodeType:
    isReachable := oc.isEgressNodeReachable(node) ==>
        func (oc *Controller) isEgressNodeReachable(egressNode *kapi.Node) bool {   ==>
             oc.isReachable(eNode)
```